src/CommonDBTM.php

```diff
@@ -5912,4 +5912,118 @@ public function getMassiveActionsForItem(): MassiveAction
 
         return new MassiveAction($params, [], 'initial', true);
     }
+
+    /**
+     * Automatically update 1-N links tables for the current item.
+     *
+     * @param string $field
+     * @param string $itemtype
+     * @param string $table
+     * @param array  $extra_input
+     *
+     * @return void
+     */
+    protected function update1NTableData(
+        string $field,
+        string $itemtype,
+        string $table,
+        array $extra_input
+    ): void {
+        global $DB;
+
+        // Compute item_1 params (current object)
+        $item_1_fk = getForeignKeyFieldForItemType(static::class);
+        $item_1_id = $this->getID();
+
+        // Compute item_2 params (linked objets(s))
+        $item_2_fk = getForeignKeyFieldForItemType($itemtype);
+
+        // Get input value
+        $input_value = $this->input[$field] ?? null;
+
+        // See dropdownField twig macro, needed for empty values as an empty
+        // array wont be sent in the HTML form
+        $input_defined_value = $this->input["_{$field}_defined"] ?? null;
+
+        // Case 1: no updates -> do nothing
+        if ($input_value === null && !$input_defined_value) {
+            return;
+        }
+
+        // Case 2: input was emptied -> remove all values
+        if (
+            ($input_value === null && $input_defined_value)
+            || (is_array($input_value) && ! count($input_value))
+        ) {
+            $DB->delete($table, $extra_input);
```

Comment:
Even `CommonDBConnexity` items may have pre/post delete operations defined in their class, so this should be done using `CommonDBTM::delete()` method.